src/analysis/useless_initializer.d

```diff
@@ -25,49 +31,142 @@ final class UselessInitializerChecker : BaseAnalyzer
 private:
 
     enum key = "dscanner.useless-initializer";
+
     version(unittest)
+    {
         enum msg = "X";
+    }
     else
+    {
         enum msg = `Variable %s initializer is useless because it does not differ from the default value`;
+    }
 
     static immutable strDefs = [`""`, `""c`, `""w`, `""d`, "``", "``c", "``w", "``d", "q{}"];
     static immutable intDefs = ["0", "0L", "0UL", "0uL", "0U", "0x0", "0b0"];
 
+    HashMap!(string, bool) _structCanBeInit;
+    DynamicArray!(string) _structStack;
+    DynamicArray!(bool) _inStruct;
+    DynamicArray!(bool) _atDisabled;
+    DynamicArray!(bool) _inTest;
+
 public:
 
     ///
     this(string fileName, bool skipTests = false)
     {
         super(fileName, null, skipTests);
+        _inStruct.insert(false);
+        _inTest.insert(false);
+    }
+
+    override void visit(const(Unittest) test)
+    {
+        _inTest.insert(true);
+        test.accept(this);
+        _inTest.removeBack();
+    }
+
+    override void visit(const(StructDeclaration) decl)
+    {
+        if (_inTest.back())
+            return;
+
+        assert(_inStruct.length > 1);
+
+        const string structName = _inStruct[$-2] ?
+            _structStack.back() ~ "." ~ decl.name.text :
+          decl.name.text;
+
+        _structStack.insert(structName);
+        _structCanBeInit[structName] = false;
+        _atDisabled.insert(false);
+        decl.accept(this);
+        _structStack.removeBack();
+        _atDisabled.removeBack();
+    }
+
+    override void visit(const(Declaration) decl)
+    {
+        _inStruct.insert(decl.structDeclaration !is null);
+        decl.accept(this);
+        if (_inStruct.length > 1 && _inStruct[$-2] && decl.constructor &&
+            ((decl.constructor.parameters && decl.constructor.parameters.parameters.length == 0) ||
+            !decl.constructor.parameters))
+        {
+            _atDisabled[$-1] = decl.attributes
+                .canFind!(a => a.atAttribute !is null && a.atAttribute.identifier.text == "disable");
+        }
```

Comment:
OT: At some point we probably should start sharing code between the checkers. I imagine sth. like `isInDisabledStruct` to be a fairly common request.